NewForce-Cohort-5/tabloidfullstack-fool-s-spring
# Atb create a post

# Description

This is a new feature which users can create a new post. Once the user fills in the form with valid details and clicks the submit button they will be redirected to view the new post they just created.

### [TICKET #6 Create a Post](https://trello.com/c/4kg2UfL6/6-6-create-a-post)

## Type of change


# Testing Instructions

- run `git add .` then `git commit` all changes in your terminal
- `git checkout main` then `git fetch -a` to get all branches that haven't been fetched yet
- `git checkout atb-post-details`
- `start ~{working-directory}/tabloidfullstack-fool-s-spring/Tabloid.sln`
- Run these queries in your SQL Object Explorer [Create DB](https://github.com/NewForce-Cohort-5/tabloidfullstack-fool-s-spring/blob/main/SQL/01_Tabloid_Create_DB.sql) and [Seed Data](https://github.com/NewForce-Cohort-5/tabloidfullstack-fool-s-spring/blob/main/SQL/02_Tabloid_Seed_Data.sql)
- Also run this query
```sql
-- Currently all of the post photo links are broken so this will generate images
UPDATE Post
SET ImageLocation = 'https://picsum.photos/920/360/'
```
- In Visual Studio, run in debugger using the `Tabloid` server
- `cd ~{working-directory}/tabloidfullstack-fool-s-spring/Tabloid/client && npm start`
- Log in as 'foo@bar.com'
- Click on the New Post link in the nav bar
- Try to create a new Post by clicking submit while fields are empty you should see the following invalidations for:
  - Title
  - Content
  - Category
- The image location as well as publication date is optional fields fill them or don't
- When the necessary fields are valid and user clicks submit they should be redirected to their post's detail page

Sample New Post
```json
{
  "id": 55,
  "title": "sample",
  "content": "Sed sagittis. Nam congue, risus semper porta volutpat, quam pede lobortis ligula, sit amet eleifend pede libero quis orci. Nullam molestie nibh in lectus.\r\n\r\nPellentesque at nulla. Suspendisse potenti. Cras in purus eu magna vulputate luctus.\r\n\r\nCum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Vivamus vestibulum sagittis sapien. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.\r\n\r\nEtiam vel augue. Vestibulum rutrum rutrum neque. Aenean auctor gravida sem.",
  "imageLocation": null,
  "createDateTime": "2019-10-02T00:00:00",
  "publishDateTime": null,
  "isApproved": true,
  "categoryId": 2,
  "category": {
    "id": 2,
    "name": "Politics"
  },
  "userProfileId": 8,
  "userProfile": {
    "id": 8,
    "firstName": "Giuseppe",
    "lastName": "Teanby",
    "displayName": "gteanby6",
    "email": "gteanby6@craigslist.orgx",
    "createDateTime": "2019-10-02T00:00:00",
    "imageLocation": "https://robohash.org/hicnihilipsa.png?size=150x150&set=set1",
    "userTypeId": 1,
    "userType": {
      "id": 1,
      "name": "Admin"
    },
    "fullName": "Giuseppe Teanby"
  }
}
```

# Checklist:
